src/test/compile-fail-fulldeps/proc-macro/cannot-link.rs

```diff
@@ -1,16 +0,0 @@
-// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
-// file at the top-level directory of this distribution and at
-// http://rust-lang.org/COPYRIGHT.
-//
-// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
-// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
-// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
-// option. This file may not be copied, modified, or distributed
-// except according to those terms.
-
-// aux-build:derive-a.rs
-
-extern crate derive_a;
-//~^ ERROR: crates of the `proc-macro` crate type cannot be linked at runtime
-
-fn main() {}
```

Comment:
There is still a phase distinction -- only the macros are available from `extern crate derive_a` and `derive_a` is not linked.

> until we actually support `use`ing derives

I was going to support this behind a feature gate in a PR tomorrow...

> I don't want the macro_use to be optional

Why not? `#[macro_use]` or the ability to `use` derives, `extern crate derive_a;` just defines an empty module `derive_a`. Maybe a warning would be appropriate, but I don't see a reason to disallow it.